helix-core/src/main/java/org/apache/helix/common/caches/BasicClusterDataCache.java

```diff
@@ -48,15 +52,50 @@
 
   public BasicClusterDataCache(String clusterName) {
     _propertyDataChangedMap = new ConcurrentHashMap<>();
-    _liveInstanceMap = new HashMap<>();
-    _instanceConfigMap = new HashMap<>();
     _externalViewCache = new ExternalViewCache(clusterName);
     _clusterName = clusterName;
-    requireFullRefresh();
+    DEFAULT_CONTEXT_PROVIDER = AbstractDataCache.createDefaultControlContextProvider(clusterName);
```

Comment:
Please refer to this class.

Reply:
@dasahcc Would you agree on this: BasicClusterDataCache implements ControlContextProvider?
ControlContextProvider is for debug use. @jiajunwang said we need the eventId when debugging. So we still need to implement ControlContextProvider. What do you think?